pkg/local_object_storage/peapod/put.go

```diff
@@ -0,0 +1,60 @@
+package peapod
+
+import (
+	"context"
+	"fmt"
+
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+)
+
+// Put saves given data in the underlying database by specified object address.
+// The data can be anything, but in practice a binary NeoFS object is expected.
+// Operation is executed within provided context: if the context is done, Put
+// returns its error (in this case data may be saved).
+func (x *Peapod) Put(ctx context.Context, addr oid.Address, data []byte) error {
+	x.currentBatchMtx.RLock()
```

Comment:
This means that either (with the current code) we have a tick-tock pattern (10ms wating then writing, then waiting/writing, etc) and don't completely utilize the disk, or (if we're to use adaptive ticker) we don't have enough changes to flush in many cases (unlock/lock happens too fast for a meaningful rlocks to happen). This will be smoothed out somewhat by the fact that the next changeset will be committed quickly, but it still may be suboptimal.

Reply:
we could try to manage batch size and/or analyze flush performance in runtime and/or analyze the input load rate, but such model is more probabilistic and requires in-depth study. 

over a long period of time, the input load should smooth out and rest against the disk, and the granular flow will rest against the 10ms interval, which is negligible. Note that right now we don't deny operations with "overflow" status.

what we can definitely add right now is the transaction reset skip in the complete absence of changes for the current interval, although this is a micro-optimization and will require additional resource (bool) requiring synchronization between threads.